src/pages/index.js

```diff
@@ -78,19 +104,129 @@ editProfileButton.addEventListener("click", () => {
   profilePopup.open();
 });
 
-//add new card
+profileImageElement.addEventListener("click", () => {
+  imageChangePopup.open();
+});
+
+/*////////////////////////////////////////////////////////////////////
+/                      Popups                                        /
+////////////////////////////////////////////////////////////////////*/
+
+const viewImagePopup = new PopupWithImage(".view-image-popup");
+const confirmDeletePopup = new PopupWithForm({
+  popupSelector: ".delete-card-popup",
+  handleFormSubmit: () => {
+    deleteCard(cardElement);
+  },
+});
 
 const addCardPopup = new PopupWithForm({
   popupSelector: ".add-card-popup",
   handleFormSubmit: (formData) => {
-    renderCard(formData);
-    addCardPopup.close();
+    addCardPopup.setSavingMessage();
+    api
+      .addCard(formData)
+      .then((data) => {
+        renderCard(data);
+        addCardPopup.close();
+        addCardPopup.resetSubmitMessage("Create");
+      })
+      .catch((err) => {
+        console.error(`Error: ${err}`);
+      });
   },
 });
 
-const addLocationButton = document.querySelector(".profile__add-button");
+const profilePopup = new PopupWithForm({
+  popupSelector: ".profile-popup",
+  handleFormSubmit: () => {
+    const data = {
+      name: editProfileName.value,
+      about: editProfileSubtitle.value,
+    };
+    profilePopup.setSavingMessage();
+    api
+      .updateUserInfo(data)
+      .then((res) => {
+        userInfo.setUserInfo({
+          name: res["name"],
+          info: res["about"],
+          image: res["avatar"],
+          id: res["_id"],
+        });
+        profilePopup.close();
+        profilePopup.resetSubmitMessage("Save");
+      })
+      .catch((err) => {
+        console.error(`Error: ${err}`);
+      });
+  },
+});
 
-addLocationButton.addEventListener("click", () => {
-  addCardPopup.open();
-  formValidators["add-location-form"].resetValidation();
+const imageChangePopup = new PopupWithForm({
+  popupSelector: ".profile-image-popup",
+  handleFormSubmit: (formData) => {
+    imageChangePopup.setSavingMessage();
+    api
+      .updateUserImage(formData["profile-image"])
+      .then((res) => {
+        imageChangePopup.close();
+        console.log("updating image");
+        profileImage.src = res.avatar;
```

Comment:
`        profileImage.src = res.avatar;
`

Class `UserInfo` should set the entire user’s data (including the **avatar**)